mlflow/resource_experiment.go

```diff
@@ -0,0 +1,125 @@
+package mlflow
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Experiment defines the parameters that can be set in the resource.
+type Experiment struct {
```

Comment:
this type must have the following definition, as general rule for TF resources is to reflect on API structure as much as possible, so that we have lesser implementation/documentation/education/support overhead. 

DTOs are still considered as intermediary abstraction layer to increase readability. Primary terraform data transfer construct is a pointer to `schema.ResourceData`.

DTOs are only needed for complex entities, like Generic Permissions, SCIM (users and groups) or SQL widgets/objects, where no direct easy mapping onto reflection-generated terraform schemas are possible. Or when there's significant configuration natural state drift for the entities, which is more difficult to fix on the backend. DTOs require additional conversion methods as well, which may be forgotten about by new developers.

MLflow entities are not complex enough to require DTO for every single entity.

```go
type Experiment struct {
	ExperimentId     string `json:"experiment_id" tf:"computed"`
	Name             string `json:"name"`
	ArtifactLocation string `json:"artifact_location,omitempty" tf:"force_new"`
	LifecycleStage   string `json:"lifecycle_stage,omitempty" tf:"computed"`
	LastUpdateTime   int64  `json:"last_update_time,omitempty" tf:"computed"`
	CreationTime     int64  `json:"creation_time,omitempty" tf:"computed"`
}
```

Reply:
done